homeassistant/components/tellstick.py

```diff
@@ -83,7 +83,7 @@ def stop_tellcore_net(event):
 
     try:
         tellcore_lib = TelldusCore(
-            callback_dispatcher=QueuedCallbackDispatcher())
+            callback_dispatcher=AsyncioCallbackDispatcher(hass.loop))
```

Comment:
The problem is, you need implement all in async to do that.

Reply:
Hmm, can you expand on that? Would that be cleaner still than flushing the queue of the QueuedCallbackDispatcher regularly (as I understood is needed to make callbacks happen for QueuedCallbackDispatcher)?  Hoping we can find some fix to this, #10329 is a pretty serious bug for my HA-usecase.

Reply:
This platform/components run in our threaded environment. So it need run inside threads. If you want run asynchronous, you need port this into our async interfaces

Reply:
I don't know the Home Assistant codebase too well, so it's not clear to me what that would require. @stefan-jonasson, ideas? @pvizeli, since you know the code well, and landed 1c8f1796903d06786060c53b48 which regressed this, perhaps you could land a fix for it too? Would love to get my switches working again 😃

Reply:
@pvizeli Can you describe the actual problem we trying to avoid (because the previous implementation worked just fine)? When you say _all_ exactly what does that entail, just the callback handling or do we need to rewrite the whole platform? If so could you point me to some examples from other platforms that implemented in that way?

Reply:
You need use the `DirectCallbackDispatcher`. I know it is realy complex to handle asyncio and threaded stuff in same code base. Short story, you can't use a threaded interface (like this component implement` inside a async context like the modul library run. I know it "work" but block our loop and make that home-assistant run into troubles.

Reply:
Create a dispatcher they implement `BaseCallbackDispatcher` and run the callback with `hass.add_job()`

Reply:
A other way is to migrate the callback handle inside this component to async and call the async interfaces from hass core

Reply:
And the last way (maybe the fast) is to make the handling with QueuedCallbackDispatcher correct and create a thread to handle the callback correct.

Reply:
Maybe we could merge this and revert to the old functionality until we have a proper fix? Seems better to have a blocking solution than one that doesn't work at all! 
I will start a new PR and see what I can do!

Reply:
I would agree with @stefan-jonasson, landing a regression-fix while a proper solution is worked on sounds like a good strategy.

Reply:
Since `AsyncioCallbackDispatcher` will affect our core loop, we can revert this change anymore. If the fix will be only affect this platform, we can do it but not the core.

Reply:
@pvizeli Are you saying we *can* or *can't* revert this? If we can't, are you saying there will be a fix for 0.57.3?

Reply:
@torarnv Until this has been resolved, you can always just modify your local tellstick.py file and revert the breaking changes yourself. I did that and my HA runs as smoothly as ever. 

Sure, there might be problems with blocking the core loop and whatnot, but I haven't experienced any issues yet.

Reply:
@stefan-jonasson we should discuss a real solution here and no flameware. If you convert the callback they will be called from tellcore library to async (tag it with @callback from core.py too) it should be work. Maybe that is the fast way and maybe also the best way.

Reply:
I give a try, but I can't test it

Reply:
@pvizeli Happy to test anything you come up with! 😄